kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/punctuators/AbstractThrottledPunctuator.java

```diff
@@ -20,12 +22,29 @@ public abstract class AbstractThrottledPunctuator<T> implements Punctuator {
   private final Clock clock;
   private final KeyValueStore<Long, List<T>> eventStore;
   private final ThrottledPunctuatorConfig config;
+  private final MeterRegistry meterRegistry;
+  private final String punctuatorName;
 
   public AbstractThrottledPunctuator(
-      Clock clock, ThrottledPunctuatorConfig config, KeyValueStore<Long, List<T>> eventStore) {
+      Clock clock,
```

Comment:
I remember commenting earlier too. This will be a breaking change if the kafka-stream-framework dependency gets upgraded wherever AbstractThrottledPuncture is already getting used.

Instead of this, add a new constructor which takes the new fields.